src/vaults.lua

```diff
@@ -1,28 +1,30 @@
-Vaults = Vaults or {}
-
 -- Utility functions that modify global Vaults object
 local vaults = {}
 local balances = require("balances")
 local utils = require("utils")
 local constants = require("constants")
 
---- @type Timestamp|nil
-NextBalanceVaultsPruneTimestamp = NextBalanceVaultsPruneTimestamp or 0
-
 --- @class Vault
---- @field balance number The balance of the vault
---- @field startTimestamp number The start timestamp of the vault
---- @field endTimestamp number The end timestamp of the vault
+--- @field balance mARIO The balance of the vault
+--- @field controller WalletAddress | nil The controller of a revokable vault. Nil if not revokable (default: nil)
+--- @field startTimestamp Timestamp The start timestamp of the vault
+--- @field endTimestamp Timestamp The end timestamp of the vault
+
+--- @alias Vaults table<WalletAddress, table<VaultId, Vault>> -- A table of vaults indexed by owner address, then by vault id
+
+--- @type Vaults
+Vaults = Vaults or {}
 
---- @class Vaults: table<string, Vault> A table of vaults indexed by owner address
+--- @type Timestamp|nil
+NextBalanceVaultsPruneTimestamp = NextBalanceVaultsPruneTimestamp or 0
 
 --- Creates a vault
 --- @param from string The address of the owner
 --- @param qty number The quantity of tokens to vault
 --- @param lockLengthMs number The lock length in milliseconds
 --- @param currentTimestamp number The current timestamp
 --- @param vaultId string The vault id
---- @return Vault The created vault
+--- @return Vault -- The created vault
```

Comment:
We've used a # elsewhere before the return explanation. Not sure if `--` also parses the same way in the IDE

Reply:
`--` and `#` result in the same IDE output for me: 
![image](https://github.com/user-attachments/assets/184b18d4-7105-4a85-88aa-a670104cbccb)